dev/tests/integration/testsuite/Magento/UrlRewrite/Controller/UrlRewriteTest.php

```diff
@@ -80,6 +80,36 @@ public function requestDataProvider(): array
                 'request' => '/page-similar/',
                 'redirect' => '/page-b',
             ],
+            'Use Case #7: Rewrite: page-similar --(301)--> page-a; '
+            . 'Request: page-similar?param=1 --(301)--> page-a?param=1' => [
+                'request' => '/page-similar?param=1',
+                'redirect' => '/page-a?param=1',
+            ],
+            'Use Case #8: Rewrite: page-similar/ --(301)--> page-b; '
+            . 'Request: page-similar/?param=1 --(301)--> page-b?param=1' => [
+                'request' => 'page-similar/?param=1',
```

Comment:
I guess all requests and redirects should be prefixed with `/`

Reply:
Fixed.